### DISCUSSION TITLE
Cruiser App functions / effects of activation of maps and POI's

### DISCUSSION THREAD

**Discussion by GuG1050** [devemux86/cruiser#189] [Q&A] [answered]
Maybe I missed the explanation somewhere in the discussions ...
My question: What happens in the Cruiser App in case I activate several different downloaded maps for the same country (e.g. mapsforge AND OpenAndroMaps AND Freizeitkarte for Austria)? Is mapsforge always the default displayed map? Is the map which is used for display controlled by the selection of the map design (e.g. Biker = mapsforge, Elevate = OpenAndroMaps)?
Similar question if mapsforge AND OpenAndroMaps POI's are activated for the sam country. Both used in the search? Depends on which map is displayed?

I am only curious because the defined behavior of the app in case of these multiple activations is not clear to me. Maybe its possible but should be avoided?
How to handle activated maps , map designs and POI's if all are of the same type I know ... (unfortunately I didn't manage to make the screenshots smaller)

<img src="https://github.com/devemux86/cruiser/assets/122791327/4450c4e1-5ef7-4b76-8943-00377c294747" width="33%"/><img src="https://github.com/devemux86/cruiser/assets/122791327/cc941967-f7e4-4a2b-9ba2-d60a4ce13750" width="33%"/><img src="https://github.com/devemux86/cruiser/assets/122791327/57973299-ba74-47d2-9c8f-bfc72e0a5334" width="33%"/>

**Comment by devemux86** [comment, marked as answer]
@GuG1050 thanks for the interesting question!

Currently when we activate multiple maps or poi, they are all used in map rendering and offline search.

But we can avoid mixing maps / poi from different sources: Mapsforge, OpenAndroMaps, Freizeitkarte.
I will see what can be done. :slightly_smiling_face: 

---

Note:

I rewrote the POI format for better offline search and more new features:
- https://github.com/devemux86/cruiser/discussions/151

Mapsforge POI are now the new version 3.
OpenAndroMaps POI are still the old version 2. So they are currently disabled in Cruiser.

Better delete the OpenAndroMaps POI in the download manager to save device storage.

---

(I edited your post to resize the images, you could check how it can be done)

To resize images in Github, there are a few tricks available. Please see here:
- https://github.com/devemux86/cruiser/discussions/27

**Reply by GuG1050** [reply]
Thank you for the answer. I will deleted the OAM poi's.
No need to change anything in the App if the activation of mixed maps and poi's from different sources does not create any disadvantages or confusion in the program.
Since updating has becomeI so convenient I normally download all maps and poi's of the countrys I like to visit and activate them all out of convenience. I simply dont do that anymore if there is any downside.

**Reply by devemux86** [reply]
It is not recommended to mix maps or poi from different sources, so I need to make some improvements here.

For example:
- Activate 3 maps of Austria from Mapsforge, OpenAndroMaps, Freizeitkarte:
map engine works x3 for the same region and may show the same data x3

- Activate 2 poi of Austria from Mapsforge, OpenAndroMaps:
offline search works on 2 databases and may return duplicate results

Although now, as explained [above](https://github.com/devemux86/cruiser/discussions/189#discussioncomment-6034533), you should all deactivate / delete the OpenAndroMaps POI.

**Comment by Ciutnik** [comment]
@devemux86 
> But we can avoid mixing maps / poi from different sources: Mapsforge, OpenAndroMaps, Freizeitkarte.

I guess we could lose something. I noticed a long time ago that using my own style (modified default Kurviger style) and at the same time activating the maps from various sources, I get the visibility of POIs of particular interest to me at higher levels of the map scale. It's as if the individual POI sets came from different maps.

I omit the issue of performance, because I did not notice a significant decrease in it.

**Reply by devemux86** [reply]
Maps from different sources may also contain different data.
That's why it's recommended to use them with their themes.

**Reply by Ciutnik** [reply]
To illustrate how map mixing looks like for me:
1 - Mapsforge (I see a gas station and a castle)
2 - OpenAndroMaps (I see a gas station, a castle and a church)
3 - Freizeitkarte (I see a gas station, a castle, a museum and a monument)
1+2+3 - allows you to see all the above POIs and all map details.

![Details](https://github.com/devemux86/cruiser/assets/122647782/6b41e3c5-ebe9-48c7-bebd-ce963712bf28)
_(for some reason the images got blurry when generating the collage, but of course the image is sharp in the app)_

**Reply by GuG1050** [reply]
That is interesting. I have to admit that I never noticed that. Wich map and which map / map design is visible in your Screenshots?

**Reply by Ciutnik** [reply]
@GuG1050 As described above, simultaneously active Mapsforge, OpenAndroMaps and Freizeitkarte maps. The map theme is the Kurviger default theme that I modified (the Cruiser equivalent is "Biker", obviously not containing my changes). These changes were made by me **very amateurishly**, in order to show museums, castles, churches and monuments on the map. As you can see, it was partially successful. As I remember, I also wanted to see historic military objects, such as bunkers, fortresses, fortifications, but I did not manage to do it - perhaps it was missing on the Mapsforge map used at the time.

**Reply by devemux86** [reply]
> perhaps it was missing on the Mapsforge map

The OSM tags included currently in Mapsforge maps can be seen [here](https://github.com/mapsforge/mapsforge/blob/master/mapsforge-map-writer/src/main/config/tag-mapping.xml).

If anyone needs more map elements / tags in maps (and themes), please suggest them and we can discuss it.

**Reply by GuG1050** [reply]
@devemux86   One more question regarding POI Download: You mentioned somewhere that the Mapsforge poi's are downloaded together with the mapsforge maps. So is it redundant to download them separately in the places of interest? 
I also understand that the Graphhopper graphs do not come together with the mapsforge maps.

I am asking because Cruiser offered to update the maps, the points of interest and the Graph Hopper Graphs for france. I did the Update of the france map, however the mapsforge poi's remained un-updated. So I could delete instead of updating them because they are double anyway ...

**Reply by devemux86** [reply]
> You mentioned somewhere that the Mapsforge poi's are downloaded together with the mapsforge maps. So is it redundant to download them separately in the places of interest?

Exactly and it is about the new downloads of maps.

When we download a new map (e.g. France map), then its POI is also automatically downloaded (e.g. France poi).
So the **offline search** can work immediately for your downloaded maps.

I have not enabled this for GraphHopper graphs (they have the same regions with maps / poi),
because many users prefer to use BRouter for offline routing.

> I did the Update of the france map, however the mapsforge poi's remained un-updated.

The updates continue to work as before, you can manually update Maps / POI / BRouter / GraphHopper.
Do we need to update the POI automatically when we update the maps?

> So I could delete instead of updating them because they are double anyway ...

What do you mean?
There are no duplicates, there is only 1 France POI.

**Reply by devemux86** [reply]
I will enable in the next Cruiser when we update the Mapsforge maps to also automatically update their POI.

**Comment by devemux86** [comment]
@GuG1050 

> I did the Update of the france map

Did you see any problem with the progress bar during the download?

Please also see this report:
- https://github.com/devemux86/cruiser/discussions/12#discussioncomment-6061294

**Reply by GuG1050** [reply]
No "problems" with the progress bar. All downloads I started finished ok so far. Maybe the bar is a bit sensitive to switching to some different App and then back during the download. Something like freezing can happen. However as everthing finally works fine I never paid much attention to the bar. In addition Android tells me when a download ist finished.
I do not close Cruiser during Downloads.

**Reply by GuG1050** [reply]
What do you mean? --> 

I did the France mapsforge map update (poi's included as I understand). But the mapsforge places of interest poi's of france remained "Update available" so presumably not updated. 

Maybe my misunderstanding that map poi's and places of interest poi's are not the same?

**Reply by devemux86** [reply]
> But the mapsforge places of interest poi's of france remained "Update available" so presumably not updated.

Automatic update of poi with map updates will be enabled in next Cruiser.
Please see my above comment: https://github.com/devemux86/cruiser/discussions/189#discussioncomment-6062685

> Maybe my misunderstanding that map poi's and places of interest poi's are not the same?

Map files and poi files have different purpose:

- Offline maps contain all the data necessary to **view maps offline**.

- Offline poi contain an extensive collection of data for **offline search** and upcoming new features:
  - https://github.com/devemux86/cruiser/discussions/9
  - https://github.com/devemux86/cruiser/discussions/195

**Comment by Ciutnik** [comment]
@devemux86 I noticed that with the last update you blocked the ability to activate multiple maps at the same time. It's a pity, because against logic and common sense I used it as described above.

Fortunately for me, you left a loophole (probably unintentionally?): you can take advantage of the fact that a newly downloaded map activates automatically, without deactivating currently active... 🙂 If you can, please don't patch it 😉

**Reply by devemux86** [reply]
@Ciutnik 

> you left a loophole (probably unintentionally?)

It was not just fully patched yet! :slightly_smiling_face: 

> I noticed that with the last update you blocked the ability to activate multiple maps at the same time.

Multiple maps and poi from the same provider should work (Mapsforge **or** OpenAndroMaps **or** Freizeitkarte).
Open 2 Germany map from 2 different providers is confusing to users and can cause problems in rendering.

I prefer to avoid it. If you used it to display POI from different maps, this was a side effect.
We should not consider it a proper solution. POI layers will be implemented in near future. :slightly_smiling_face:

**Reply by Ciutnik** [reply]
As you've noticed, I'm using the map overlay side effect to get the POI. I agree that it is not very fortunate and recommended. Waste of resources as well as some POIs are naturally duplicated. 

I am looking forward to advanced POI support, for which I thank you very much in advance 👍
